aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/GetObjectRequest.java

```diff
@@ -1008,4 +1008,47 @@ public GetObjectRequest withS3ObjectId(S3ObjectId s3ObjectId) {
         setS3ObjectId(s3ObjectId);
         return this;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) return true;
+        if (object == null || getClass() != object.getClass()) return false;
+        if (!super.equals(object)) return false;
+
+        GetObjectRequest that = (GetObjectRequest) object;
+
+        if (isRequesterPays() != that.isRequesterPays()) return false;
+        if (!s3ObjectIdBuilder.equals(that.s3ObjectIdBuilder)) return false;
+        if (!java.util.Arrays.equals(getRange(), that.getRange())) return false;
+        if (!getMatchingETagConstraints().equals(that.getMatchingETagConstraints())) return false;
+        if (!nonmatchingEtagConstraints.equals(that.nonmatchingEtagConstraints)) return false;
```

Comment:
Is there a reason you sometimes reference the property directly `!nonmatchingEtagConstraints.equals(that.nonmatchingEtagConstraints)` and you sometimes use the getter `getModifiedSinceConstraint() != null ? !getModifiedSinceConstraint().equals(that.getModifiedSinceConstraint()` ?